src/PaymentServer/Redemption.hs

```diff
@@ -85,12 +91,23 @@ import PaymentServer.Issuer
   , Issuer
   )
 
+-- | Parameters controlling exactly how vouchers are redeemed for tokens.
+data RedemptionConfig
+  = RedemptionConfig
+  { redemptionConfigNumGroups :: Int
+  -- ^ The number of redemption groups a single redemption is divided into.
+  , redemptionConfigTokensPerVoucher :: Int
+  -- ^ The total number of tokens received in exchange for one voucher.
+  , redemptionConfigIssue :: Issuer
+  -- ^ An issuer which can actually issue signatures during a redemption.
+  }
+
 data Result
   = Unpaid -- ^ A voucher has not been paid for.
   | DoubleSpend -- ^ A voucher has already been redeemed.
   | OtherFailure Text -- ^ Some other unrecognized failure mode.
   -- | Given counter was not in the expected range
-  | CounterOutOfBounds Integer Integer Integer
+  | CounterOutOfBounds Int Int Int
```

Comment:
Not really related to this PR, but I noticed it while trying to understand it:

I don't think any of these values need to be a part of `Result`, as we never return them as values of type `Result` (but rather encode them as part of `ServerError`). I'm not sure how it should be structured, though [this](https://docs.servant.dev/en/stable/cookbook/uverb/UVerb.html) has some discussion of various possibilities, including some packages that help with it.